### PR TITLE
Fix remaining check warnings and undeclared dependencies from PEcAn.allometry

### PR DESCRIPTION
**Pull request by moki1202**
Fix remaining check warnings and undeclared dependencies from PEcAn.allometry

### CODE REVIEW

File: modules/allometry/inst/AllomDriver.R
Comment (infotroph) [inline modules/allometry/inst/AllomDriver.R:17]: Let's leave these in place -- library calls OK inside `inst/` scripts

**Comment by moki1202** [comment]
@infotroph are there any other changes required here?

File: modules/allometry/DESCRIPTION
Comment (infotroph) [inline modules/allometry/DESCRIPTION:6]:
```suggestion
Authors@R: c(person("Mike","Dietze", role = c("aut","cre"),
                                       email = "dietze@bu.edu"),
```

File: modules/allometry/DESCRIPTION
Comment (infotroph) [inline modules/allometry/DESCRIPTION:6]:
```suggestion
Authors@R: c(person("Mike", "Dietze", role = c("aut", "cre"),
```

**Comment by mdietze** [comment]
@infotroph @moki1202 this PR looks like it's in pretty good shape. I'm happy to pull it in as soon as the GH Action tests pass. Are there other things y'all wanted to integrate into this PR before it is merged? Currently failing on pecan.depends.R

**Comment by infotroph** [comment]
@mdietze @moki1202 I agree this looks good -- if there was anything else that needed to happen, I don't remember it now!

**Comment by moki1202** [comment]
@mdietze @infotroph Let me just check if I can remember if this required some changes or not.

**Comment by moki1202** [comment]
@infotroph Since I can't access the VM, could you please run the `R CMD` checks on allometry and let me know the check results? Right now I see a lot of `no visible binding errors` in the file, so just wanted to check.

**Review by infotroph — APPROVED** [review]
I'm not sure why CI isn't running, but I'm approving now in case that unsticks something

File: modules/allometry/DESCRIPTION
Comment (infotroph) [inline modules/allometry/DESCRIPTION:8]:
```suggestion
             person("Shashank", "Singh", role = c("ctb")),
```

**Comment by mdietze** [comment]
Please take a look at the failures in both test and check, which bring up two separate issues. On the file-size one, I wonder if you could force the docs to build using lower-res images, or even just force the graphics to be raster not vector. If not, I think I had earlier suggestions on slack for what plots could be cut or made optional (so that one could run them manually but they wouldn't knit into the vignette)

**Comment by Aariq** [comment]
> Please take a look at the failures in both test and check, which bring up two separate issues. On the file-size one, I wonder if you could force the docs to build using lower-res images, or even just force the graphics to be raster not vector. If not, I think I had earlier suggestions on slack for what plots could be cut or made optional (so that one could run them manually but they wouldn't knit into the vignette)

It's not the images, it's the hundreds of warnings like this:
```
## Warning in u2 + 0.5 * crossprod(Y - X %*% beta): Recycling array of length 1 in vector-array arithmetic is deprecated.
##   Use c() or as.vector() instead.
```

If I turn off all the output except the plots the rendered vignette is 867.4kb

File: modules/allometry/R/allom.BayesFit.R
Comment (mdietze) [inline modules/allometry/R/allom.BayesFit.R:263]:
```suggestion
        u2 <- u2 + as.numeric(0.5 * crossprod(Y - X %*% beta))
```

Comment (mdietze) [reply modules/allometry/R/allom.BayesFit.R:263]: @Aariq I think this will fix the warning you identified. Is this the only one or were there other similar warnings?

**Comment by infotroph** [comment]
Thanks @Aariq for the diagnostics and @mdietze for the fix! Checks are now passing and this is ready to merge.

**Pull request merged**
